mdsuite/transformations/transformations.py

```diff
@@ -267,30 +271,50 @@ def _save_coordinates(
         -------
         saves the tensor_values to the database_path.
         """
-        try:
-            self.database.add_data(
-                data=data,
-                structure=data_structure,
-                start_index=index + self.offset,
-                batch_size=batch_size,
-                system_tensor=system_tensor,
-                tensor=tensor,
+
+        # turn data into trajectory chunk
+        # data_structure is dict {'/path/to/property':{'indices':irrelevant, 'columns':deduce->deduce n_dims, 'length':n_particles}
+        species_list = list()
+        # data structure only has 1 element
+        key, val = list(data_structure.items())[0]
+        path = str(copy.copy(key))
+        path.rstrip("/")
+        path = path.split("/")
+        prop_name = path[-1]
+        sp_name = path[-2]
+        n_particles = val.get("length", 1)
+        if len(np.shape(data)) == 2:
+            # data not for multiple particles, instead one value for all
+            # -> create the n_particle axis
+            data = data[np.newaxis, :, :]
+        prop = mdsuite.database.simulation_database.PropertyInfo(
+            name=prop_name, n_dims=len(val["columns"])
+        )
+        species_list.append(
+            mdsuite.database.simulation_database.SpeciesInfo(
+                name=sp_name, properties=[prop], n_particles=n_particles
             )
+        )
+        chunk = mdsuite.database.simulation_database.TrajectoryChunkData(
+            chunk_size=batch_size, species_list=species_list
+        )
+        # data comes from transformation with time in 1st axis, add_data needs it in 0th axis
+        chunk.add_data(
+            data=np.swapaxes(data, 0, 1),
+            config_idx=0,
+            species_name=sp_name,
+            property_name=prop_name,
+        )
+
+        try:
+            self.database.add_data(chunk=chunk, start_idx=index + self.offset)
         except OSError:
             """
             This is used because in Windows and in WSL we got the error that
             the file was still open while it should already be closed. So, we
             wait, and we add again.
             """
```

Comment:
I don't know but I have the feeling that `time.sleep(0.5)` was important here.

Reply:
I think I added that...
I had some issues with file handling in Windows and adding this small delay fixed it. 
We can try without it again, maybe with the refactoring it is not needed anymore

Reply:
woops sorry I did not mean to remove it